common/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php

```diff
@@ -75,6 +75,17 @@ public function validate(\Nethgui\Controller\ValidationReportInterface $report)
         ) {
             $report->addValidationErrorMessage($this, 'domain', 'valid_relay_notprimarydomain');
         }
+
+        // Domain validation, it must not be named like the first domain (FQDN)
+        $domainFQDN = $this->getPlatform()->getDatabase('configuration')->getType('DomainName');
+        $domain = $this->parameters['domain'];
+
+        if($this->getRequest()->isMutation()
+            && preg_match ("/$domainFQDN/", $domain)
+            && $this->parameters['TransportType'] === 'LocalDelivery'
```

Comment:
The `TransportType` shouldn't be checked. I guess FQDN is never acceptable because it's already defined internally by Postfix.

Reply:
Are you sure ? even if the FQDN of this server is domain.com, why I cannot forward all emails to sub.domain.com ?

Reply:
In that case we must forbid "domain.com" itself. "sub.domain.com" is acceptable.

"domain.com" cannot be set as host name though, because IIRC two dots are required by FQDN field.